common-pg/src/main/scala/com/socrata/pg/soql/SoQLAnalysisSqlizer.scala

```diff
@@ -96,10 +96,16 @@ object BinarySoQLAnalysisSqlizer extends Sqlizer[(BinaryTree[SoQLAnalysis[UserCo
     analyses match {
       case PipeQuery(l, Leaf(ra)) =>
         val (lpsql, _) = sql(l, tableNames, allColumnReps, reqRowCount, rep, typeRep, setParams, ctx, escape, fromTableName)
-        val chainedTableAlias = "x1"
-        val subTableName = "(%s) AS %s".format(lpsql.sql.head, chainedTableAlias)
-        val tableNamesSubTableNameReplace = tableNames + (TableName.PrimaryTable -> subTableName)
-        val primaryTableAlias = if (ra.joins.nonEmpty) Map((PrimaryTableAlias -> chainedTableAlias)) else Map.empty
+        val (primaryTableName, chainedTableAlias) = ra.from match {
+            case Some(tn@TableName(name, Some(alias))) if name == TableName.This =>
+              (tn.copy(alias = None), alias)
+            case _ =>
+              (TableName.PrimaryTable, "x1")
+          }
+        val subTableName = if (ra.from.isEmpty) "(%s) AS %s".format(lpsql.sql.head, chainedTableAlias)
+                           else "(%s)".format(lpsql.sql.head)
+        val tableNamesSubTableNameReplace = tableNames + (primaryTableName -> subTableName)
+        val primaryTableAlias = if (ra.joins.nonEmpty) Map((PrimaryTableAlias -> chainedTableAlias)) else Map.empty // REVISIT
```

Comment:
But if the user has aliased `this` during a join, we do use the provided alias? Looks like this was always defaulting to `x1` before this change--what does `x1` refer to?

Reply:
"x1" is the implicit (unused) alias we use for subquery (chained soql).  And it isn't new.

Reply:
select c |> select c as c2 
will look like
select c as c3 from (select c) as x1